### PR TITLE
[Cart] Add Validator for localeCode

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         | 1.10
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Validate localeCode for picking up cart endpoint.
image

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/validator.xml [hunk 1]
```diff
@@ -90,6 +90,11 @@
             <tag name="validator.constraint_validator" alias="sylius_api_confirm_reset_password" />
         </service>
 
+        <service id="sylius.api.validator.pickup_cart" class="Sylius\Bundle\ApiBundle\Validator\Constraints\PickupCartLocaleValidator">
```
Comment (Ferror) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/validator.xml:93]:
```suggestion
        <service id="sylius.api.validator.pickup_cart_locale" class="Sylius\Bundle\ApiBundle\Validator\Constraints\PickupCartLocaleValidator">
```

File: src/Sylius/Bundle/ApiBundle/Resources/translations/validators.en.yaml [hunk 2]
```diff
@@ -7,6 +7,8 @@ sylius:
         without_country: 'The address without country cannot exist'
     country:
         not_exist: 'The country %countryCode% does not exist.'
+    locale:
+        not_exist: 'The locale %localeCode% does not exist.'
```
Comment (Ferror) [inline src/Sylius/Bundle/ApiBundle/Resources/translations/validators.en.yaml:11]: How about we provide an example.. like:
```suggestion
        not_exist: 'The locale %localeCode% does not exist. Try en_US'
```

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Validator/Constraints/PickupCartLocaleValidator.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        /** @var ChannelInterface $channel */
+        $channel = $this->channelRepository->findOneByCode($value->getChannelCode());
+
+        $locales = $channel->getLocales();
+
+        $locale = $locales->filter(function (LocaleInterface $locale) use ($value): bool {
+            return $locale->getCode() === $value->localeCode;
+        });
+
+        if ($locale->isEmpty()) {
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Validator/Constraints/PickupCartLocaleValidator.php:47]:
```suggestion
        $locales = $channel->getLocales();
        $locales = $locales->filter(function (LocaleInterface $locale) use ($value): bool {
            return $locale->getCode() === $value->localeCode;
        });

        if ($locales->isEmpty()) {
```

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/translations/validators.en.yaml (hunk 2 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/ApiBundle/Resources/translations/validators.en.yaml:11]: We could put the parameter into validation error and plug default or all available locales into it

```suggestion
        not_exist: 'The locale %localeCode% does not exist. Try one of %possibleLocaleCodes%'
```

File: src/Sylius/Bundle/ApiBundle/Validator/Constraints/PickupCartLocaleValidator.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+        /** @var ChannelInterface $channel */
+        $channel = $this->channelRepository->findOneByCode($value->getChannelCode());
+
+        $locales = $channel->getLocales()->filter(function (LocaleInterface $locale) use ($value): bool {
+            return $locale->getCode() === $value->localeCode;
+        });
+
+        if ($locales->isEmpty()) {
+            $this->context->addViolation($constraint->notExist, ['%localeCode%' => $value->localeCode]);
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Validator/Constraints/PickupCartLocaleValidator.php:50]: If we decide to add possible locale codes, it should be done here

**Review by lchrusciel — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, @SirDomin! :1st_place_medal:

**Pull request merged**